File: backend/BusinessService/Common/LogicService/WebInterface/VM/CmLogicVM.cs
// *******************************************************************
// 業務名称　: 健康管理システム
// 機能概要　: ロジック共通仕様処理
//             ViewModel定義
// 作成日　　: 2023.07.18
// 作成者　　: 劉
// 変更履歴　:
// *******************************************************************
using Newtonsoft.Json;

namespace BCC.Affect.Service.Common
{
    /// <summary>
    /// 権限モデル
    /// </summary>
    public class CmAuthVM
    {
        public bool personalflg { get; set; }           //個人番号操作権限フラグ
        public bool alertviewflg { get; set; }          //警告参照フラグ
        public List<string> editSisyos { get; set; }    //更新可能支所一覧
    }

    /// <summary>
    /// 支所モデル
    /// </summary>
    public class CmSisyoVM
    {
        public string sisyocd { get; set; } //支所コード
        public string sisyonm { get; set; } //支所名
    }

    /// <summary>
    /// ヘッダー基本情報
    /// </summary>
    public class CmHeaderBaseVM
    {
        public string name { get; set; }            //氏名
        public string kname { get; set; }           //カナ氏名
        public string sex { get; set; }             //性別
        public string juminkbn { get; set; }        //住民区分
        public string bymd { get; set; }            //生年月日
    }

    /// <summary>
    /// 共通バー共通基本情報
    /// </summary>
    public class CommonBarHeaderBaseVM : CmHeaderBaseVM
    {
        public string age { get; set; }             //年齢
        public string agekijunymd { get; set; }     //年齡計算基準日
    }

    /// <summary>
    /// 共通バー共通基本情報
    /// </summary>
    public class CommonBarHeaderBase2VM : CommonBarHeaderBaseVM
    {
        public string keikoku { get; set; } //警告内容
    }

    /// <summary>
    /// 共通バー共通基本情報
    /// </summary>
    public class CommonBarHeaderBase3VM : CommonBarHeaderBase2VM
    {
        public string adrs { get; set; }    //住所
    }

    /// <summary>
    /// 画面共通基本情報
    /// </summary>
    public class GamenHeaderBaseVM : CommonBarHeaderBaseVM
    {
        [JsonIgnore]
        public string personalno { get; set; }  //個人番号
    }

    /// <summary>
    /// 画面共通基本情報
    /// </summary>
    public class GamenHeaderBase2VM : GamenHeaderBaseVM
    {
        public string adrs { get; set; }    //住所
        public string keikoku { get; set; } //警告内容
    }

    /// <summary>
    /// 基準値範囲
    /// </summary>
    public class KjItemVM
    {
        public string itemcd { get; set; }           //項目コード
        public string kijunvaluehyoki { get; set; }  //基準値表記
        public string alertvaluehyoki { get; set; }  //注意値表記
        public string errvaluehyoki { get; set; }    //異常値表記
        public List<KijunVM> kijunlist { get; set; } //基準値リスト
    }

    /// <summary>
    /// 基準値(チェック用)
    /// </summary>
    public class KijunVM
    {
        public double? errvalue1t { get; set; }      //異常値（数値）以下
        public double? alertvalue1f { get; set; }    //注意値（数値）開始
        public double? alertvalue1t { get; set; }    //注意値（数値）終了
        public double? kijunvaluef { get; set; }     //基準値（数値）開始
        public double? kijunvaluet { get; set; }     //基準値（数値）終了
        public double? alertvalue2f { get; set; }    //注意値（数値）開始
        public double? alertvalue2t { get; set; }    //注意値（数値）終了
        public double? errvalue2f { get; set; }      //異常値（数値）以上
        public string hanteicd { get; set; }         //基準値（コード）
        public string alerthanteicd { get; set; }    //注意値（コード）
        public string errhanteicd { get; set; }      //異常値（コード）
    }
    /// <summary>
    /// フリー項目モデル(初期化)：フリー
    /// </summary>
    public class FreeItemBaseVM : FixFreeItemBaseVM
    {
        public string? groupid2 { get; set; }          //グループID2
    }
    /// <summary>
    /// フリー項目モデル(初期化)：固定
    /// </summary>
    public class FixFreeItemBaseVM : FreeItemUpdBaseVM
    {
        public string itemnm { get; set; }                      //項目名
        public Enum画面項目入力 inputkbn { get; set; }          //画面項目入力区分
        public List<DaSelectorModel> cdlist { get; set; }       //一覧選択リスト
        public int? keta1 { get; set; }                         //入力桁(整数部/文字)
        public int? keta2 { get; set; }                         //入力桁(小数部)
        public bool hissuflg { get; set; }                      //必須フラグ
        public string hanif { get; set; }                       //入力範囲（開始）
        public string hanit { get; set; }                       //入力範囲（終了）
        public bool inputflg { get; set; }                      //入力フラグ
        public EnumMsgCtrlKbn msgkbn { get; set; }              //メッセージ区分
        public string syokiti { get; set; }                     //初期値
        public string biko { get; set; }                        //備考
    }
    /// <summary>
    /// フリー項目モデル(更新)
    /// </summary>
    public class FreeItemUpdBaseVM
    {
        public string itemcd { get; set; }                  //項目コード
        public object? value { get; set; }                  //値
        public bool? fusyoflg { get; set; }                 //不詳フラグ
        public Enum入力タイプ inputtypekbn { get; set; }    //入力タイプ区分
    }
    /// <summary>
    /// 一覧列情報
    /// </summary>
    public class ColumnInfoVM
    {
        public string title { get; set; }   //項目論理名
        public string key { get; set; }     //項目物理名
    }
    /// <summary>
    /// 一覧データ
    /// </summary>
    public class DataInfoVM
    {
        public string key { get; set; }     //項目物理名
        public string value { get; set; }   //値
    }
}

File: GJSBackend/BusinessService/Common/LogicService/WebInterface/VM/CmLogicVM.vb
' *******************************************************************
' 業務名称　: 互助防疫システム
' 機能概要　: ロジック共通仕様処理
'             ViewModel定義
' 作成日　　: 2023.07.18
' 作成者　　: 
' 変更履歴　:
' *******************************************************************
Imports Newtonsoft.Json

Namespace Jbd.Gjs.Service.Common
    ''' <summary>
    ''' 権限モデル
    ''' </summary>
    Public Class CmAuthVM
        Public Property personalflg As Boolean           '個人番号操作権限フラグ
        Public Property alertviewflg As Boolean          '警告参照フラグ
        Public Property editSisyos As List(Of String)    '更新可能支所一覧
    End Class

    ''' <summary>
    ''' 支所モデル
    ''' </summary>
    Public Class CmSisyoVM
        Public Property sisyocd As String '支所コード
        Public Property sisyonm As String '支所名
    End Class

    ''' <summary>
    ''' ヘッダー基本情報
    ''' </summary>
    Public Class CmHeaderBaseVM
        Public Property name As String            '氏名
        Public Property kname As String           'カナ氏名
        Public Property sex As String             '性別
        Public Property juminkbn As String        '住民区分
        Public Property bymd As String            '生年月日
    End Class

    ''' <summary>
    ''' 共通バー共通基本情報
    ''' </summary>
    Public Class CommonBarHeaderBaseVM
        Inherits CmHeaderBaseVM
        Public Property age As String             '年齢
        Public Property agekijunymd As String     '年齡計算基準日
    End Class

    ''' <summary>
    ''' 共通バー共通基本情報
    ''' </summary>
    Public Class CommonBarHeaderBase2VM
        Inherits CommonBarHeaderBaseVM
        Public Property keikoku As String '警告内容
    End Class

    ''' <summary>
    ''' 共通バー共通基本情報
    ''' </summary>
    Public Class CommonBarHeaderBase3VM
        Inherits CommonBarHeaderBase2VM
        Public Property adrs As String    '住所
    End Class

    ''' <summary>
    ''' 画面共通基本情報
    ''' </summary>
    Public Class GamenHeaderBaseVM
        Inherits CommonBarHeaderBaseVM
        <JsonIgnore>
        Public Property personalno As String  '個人番号
    End Class

    ''' <summary>
    ''' 画面共通基本情報
    ''' </summary>
    Public Class GamenHeaderBase2VM
        Inherits GamenHeaderBaseVM
        Public Property adrs As String    '住所
        Public Property keikoku As String '警告内容
    End Class

    ''' <summary>
    ''' 基準値範囲
    ''' </summary>
    Public Class KjItemVM
        Public Property itemcd As String           '項目コード
        Public Property kijunvaluehyoki As String  '基準値表記
        Public Property alertvaluehyoki As String  '注意値表記
        Public Property errvaluehyoki As String    '異常値表記
        Public Property kijunlist As List(Of KijunVM) '基準値リスト
    End Class

    ''' <summary>
    ''' 基準値(チェック用)
    ''' </summary>
    Public Class KijunVM
        Public Property errvalue1t As Double      '異常値（数値）以下
        Public Property alertvalue1f As Double    '注意値（数値）開始
        Public Property alertvalue1t As Double    '注意値（数値）終了
        Public Property kijunvaluef As Double     '基準値（数値）開始
        Public Property kijunvaluet As Double     '基準値（数値）終了
        Public Property alertvalue2f As Double    '注意値（数値）開始
        Public Property alertvalue2t As Double    '注意値（数値）終了
        Public Property errvalue2f As Double      '異常値（数値）以上
        Public Property hanteicd As String         '基準値（コード）
        Public Property alerthanteicd As String    '注意値（コード）
        Public Property errhanteicd As String      '異常値（コード）
    End Class
    ''' <summary>
    ''' フリー項目モデル(初期化)：フリー
    ''' </summary>
    Public Class FreeItemBaseVM
        Inherits FixFreeItemBaseVM
        Public Property groupid2 As String          'グループID2
    End Class
    ''' <summary>
    ''' フリー項目モデル(初期化)：固定
    ''' </summary>
    Public Class FixFreeItemBaseVM
        Inherits FreeItemUpdBaseVM
        Public Property itemnm As String                      '項目名
        Public Property inputkbn As Enum画面項目入力          '画面項目入力区分
        Public Property cdlist As List(Of Db.DaSelectorModel)       '一覧選択リスト
        Public Property keta1 As Integer                         '入力桁(整数部/文字)
        Public Property keta2 As Integer                         '入力桁(小数部)
        Public Property hissuflg As Boolean                      '必須フラグ
        Public Property hanif As String                       '入力範囲（開始）
        Public Property hanit As String                       '入力範囲（終了）
        Public Property inputflg As Boolean                      '入力フラグ
        Public Property msgkbn As Db.EnumMsgCtrlKbn              'メッセージ区分
        Public Property syokiti As String                     '初期値
        Public Property biko As String                        '備考
    End Class
    ''' <summary>
    ''' フリー項目モデル(更新)
    ''' </summary>
    Public Class FreeItemUpdBaseVM
        Public Property itemcd As String                  '項目コード
        Public Property value As Object                  '値
        Public Property fusyoflg As Boolean                 '不詳フラグ
        Public Property inputtypekbn As Db.Enum入力タイプ    '入力タイプ区分
    End Class
    ''' <summary>
    ''' 一覧列情報
    ''' </summary>
    Public Class ColumnInfoVM
        Public Property title As String   '項目論理名
        Public Property key As String     '項目物理名
    End Class
    ''' <summary>
    ''' 一覧データ
    ''' </summary>
    Public Class DataInfoVM
        Public Property key As String     '項目物理名
        Public Property value As String   '値
    End Class
End Namespace
